lib/frr_pthread.c

```diff
@@ -99,10 +99,14 @@ struct frr_pthread *frr_pthread_new(struct frr_pthread_attr *attr)
 		if (!hash_lookup(frr_pthread_hash, &holder)) {
 			fpt = XCALLOC(MTYPE_FRR_PTHREAD,
 				      sizeof(struct frr_pthread));
+			/* initialize mutex */
+			pthread_mutex_init(&fpt->mtx, NULL);
 			/* create new thread master */
-			fpt->master = thread_master_create(attr->name);
+			fpt->master = thread_master_create(name);
 			/* set attributes */
 			fpt->attr = *attr;
+			name = (name ? name : "Anonymous thread");
```

Comment:
Should we require an actual name passed in for all pthreads?

Reply:
Probably not. For the pthreads we have so far they are 'fat' threads in the sense that they are long-lived and act as mini task servers that take care of a specific set of functionality for the life of the program (e.g. keepalives for all of bgp), and names make sense for those. But the other model is spinning up a bunch of small threads that do their work and then die (fork-join) and naming these doesn't make much sense as they usually won't live more than a second.